src/VisualStudio/Core/Impl/SolutionExplorer/DiagnosticItem/DiagnosticItem.cs

```diff
@@ -30,29 +29,16 @@ internal sealed partial class DiagnosticItem(
 
     public ProjectId ProjectId { get; } = projectId;
     public DiagnosticDescriptor Descriptor { get; } = descriptor;
-    public ReportDiagnostic EffectiveSeverity { get; private set; } = effectiveSeverity;
-
-    public override event PropertyChangedEventHandler? PropertyChanged;
+    private readonly ReportDiagnostic _effectiveSeverity = effectiveSeverity;
 
     public override ImageMoniker IconMoniker
-        => MapEffectiveSeverityToIconMoniker(this.EffectiveSeverity);
+        => MapEffectiveSeverityToIconMoniker(_effectiveSeverity);
 
     public override IContextMenuController ContextMenuController => _commandHandler.DiagnosticContextMenuController;
 
     public override object GetBrowseObject()
         => new BrowseObject(this);
 
-    internal void UpdateEffectiveSeverity(ReportDiagnostic newEffectiveSeverity)
```

Comment:
we no longer update these in place (it makes equality/hash semantics non-ideal, which does impact how the observable collection and virtual tree view operate).

instead, if things do change (the rare case) we rebuild the collection, with all the data immutable in it.

Reply:
💭 Will this cause the solution explorer to collapse nodes that are expanded when the children change? We've seen bugs of that form in other contexts before, so just asking.

Reply:
missed thsi.  the answer is likely 'yes'.  If we feel like that's important to address, we can potentially do it by breaking out the portion of the anlaysis to determine if we think an actual change to the *list* happened, versus a property change to an *item* in the list.  that said.  only the item itself collapses.  not the parent.  so it's notthat bad imo.